scripts/wsmux.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# The wsmux command makes it easy to switch between catkin workspaces. This
+# is mainly intended for use on vehicles as each user will have their own
+# workspace containing repositories that reflect their branches. In the event
+# that they need to test unmerged code in the field, they will run the node
+# from their workspace. It can also be used to manage multiple workspaces on
+# user machines, if you're into that sort of thing.
+
+
+WS_MARKER=".catkin_workspace"
+
+
+_catkin_ws_complete() {
+	local FILE
+
+	# Check for workspaces in the home directory based on the argument
+	for FILE in ~/"$2"*; do
+
+		# Skip any directory that does not contain the marker file
+		[[ -f $FILE/$WS_MARKER ]] || continue
+
+		# Append just the name of the workspace folder to the autocomplete list
+		COMPREPLY+=( `echo "$FILE" | rev | cut -d "/" -f1 | rev` )
+	done
+}
```

Comment:
I'm not a fan of leaving marker files scattered around. If there is another alternative, I would be happy to see it explored. For now, I'm ok giving this a shot.